src/loaders/assetLoader.js

```diff
@@ -129,7 +130,7 @@ module.exports = async function assetLoader() {
     this.emitFile(dest, item.content);
   });
 
-  let publicPath = `__webpack_public_path__ + ${JSON.stringify(path.join('/', assets, url))}`;
+  let publicPath = `__webpack_public_path__ + ${JSON.stringify(path.join('/', assets, url).replace(pathSepPattern, '/'))}`;
```

Comment:
I don't get your point. The `path.join` would still use the platform specific `path.sep`, but react-native code expects it to be / (since it's from a URL), see: https://github.com/facebook/react-native/blob/1fa95ed390d96b4a54702acdf89594af1785a3fd/local-cli/bundle/assetPathUtils.js#L38